crates/distribution-types/src/prioritized_distribution.rs

```diff
@@ -254,41 +243,37 @@ impl<'a> CompatibleDist<'a> {
             } => source_dist,
         }
     }
+}
 
-    /// Return the [`Yanked`] status of the distribution.
-    ///
-    /// It is possible for files to have a different yank status per PEP 592 but in the official
-    /// PyPI warehouse this cannot happen.
-    ///
-    /// Here, we will treat the distribution is yanked if the file we will install with
-    /// is yanked.
-    ///
-    /// PEP 592: <https://peps.python.org/pep-0592/#warehouse-pypi-implementation-notes>
-    pub fn yanked(&self) -> &Yanked {
-        &self.for_installation().yanked
+impl WheelCompatibility {
+    pub fn is_compatible(&self) -> bool {
+        matches!(self, Self::Compatible(_))
     }
-}
 
-impl Ord for WheelCompatibility {
-    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+    pub fn is_higher_priority(&self, other: &WheelCompatibility) -> bool {
         match (self, other) {
-            (Self::Compatible(p_self), Self::Compatible(p_other)) => p_self.cmp(p_other),
-            (Self::Incompatible(_), Self::Compatible(_)) => std::cmp::Ordering::Less,
-            (Self::Compatible(_), Self::Incompatible(_)) => std::cmp::Ordering::Greater,
-            (Self::Incompatible(t_self), Self::Incompatible(t_other)) => t_self.cmp(t_other),
+            (Self::Compatible(_), Self::Incompatible(_)) => true,
+            (Self::Compatible(tag_priority), Self::Compatible(other_tag_priority)) => {
+                tag_priority > other_tag_priority
+            }
+            (Self::Incompatible(_), Self::Compatible(_)) => false,
+            (Self::Incompatible(incompatibility), Self::Incompatible(other_incompatibility)) => {
+                incompatibility.is_more_compatible(other_incompatibility)
+            }
         }
     }
 }
 
-impl PartialOrd for WheelCompatibility {
-    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
-        Some(Self::cmp(self, other))
-    }
-}
-
-impl WheelCompatibility {
-    pub fn is_compatible(&self) -> bool {
-        matches!(self, Self::Compatible(_))
+impl SourceDistCompatibility {
+    pub fn is_higher_priority(&self, other: &SourceDistCompatibility) -> bool {
+        match (self, other) {
+            (Self::Compatible, Self::Incompatible(_)) => true,
+            (Self::Compatible, Self::Compatible) => false, // Arbitary
```

Comment:
Is that still deterministic, e.g. if the order we get from the index changes?

Reply:
It's not a change from the current status-quo where we take the "first" compatible source distribution. I think we sort the output from the index immediately though (as a guard against non-determinism).